src/clr-angular/data/datagrid/datagrid-detail.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import { Component, Inject, OnDestroy, PLATFORM_ID, Renderer2 } from '@angular/core';
+import { DetailService } from './providers/detail.service';
+import { ESC } from '../../utils/key-codes/key-codes';
+import { isPlatformBrowser } from '@angular/common';
+
+@Component({
+  selector: 'clr-dg-detail',
+  host: {
+    '[class.datagrid-detail-pane]': 'true',
+  },
+  // We put the *ngIf on the clrFocusTrap so it doesn't always exist on the page
+  template: `
+    <div clrFocusTrap class="datagrid-detail-pane-content" *ngIf="detailService.isOpen">
+      <ng-content></ng-content>
+    </div>
+    `,
+})
+export class ClrDatagridDetail implements OnDestroy {
+  private listener: () => void;
+
+  // For some reason was not getting the HostListener to fire in tests...need to investigate
+  constructor(
+    public detailService: DetailService,
+    private renderer: Renderer2,
+    @Inject(PLATFORM_ID) private platformId: Object
+  ) {
+    this.listener = this.renderer.listen('document', 'keydown', event => {
+      if (event.keyCode === ESC) {
```

Comment:
Based on my comment, HostListener wasn't working in tests but this was. I will look into it again though.

Reply:
Also keyCode is deprecated but code is not supported on IE11.

Reply:
IIRC `KeyboardEvent.key` is the one that works across Browsers with minimal inconsistencies. But Angular Pseudo Elements is there for removing this kind of quirky browser supports. Also, this not working in the tests might be because of how we implemented this `createKeyboardEvent` utility method. But It's tricky to emulate keyboard events, to say the least:
https://stackoverflow.com/questions/31491549/document-dispatchevent-not-working-in-firefox
https://stackoverflow.com/questions/8942678/keyboardevent-in-chrome-keycode-is-0/12522752#12522752